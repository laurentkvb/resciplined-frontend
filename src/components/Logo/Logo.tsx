import React from "react";

import logo from "@assets/logo-large.png";
import { SystemProps } from "@models/styledSystem";
import styled from "styled-components";

interface Props extends SystemProps {
  src?: string;
  width?: number;
}

const LogoImage: React.FC<Props> = styled.img<Props>`
filter: invert(100%)`; // instead of black color, logo should have white color

const Logo = ({ width }: Props) => (
  <LogoImage
    src={logo}
    width={width || undefined}
    data-testid="Logo"
  />
);

export default Logo;
