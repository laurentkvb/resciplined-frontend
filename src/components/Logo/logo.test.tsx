import React from "react";
import { render } from "../../tests/test-utils";

import Logo from "./Logo";

test("Logo snapshot test", () => {
  const { container } = render(<Logo />);

  // Asserts Logo component matches snapshot.
  expect(container.firstChild).toMatchSnapshot();
});

test("Logo renders as img", () => {
  const { getByTestId } = render(<Logo />);
  const renderedLogo = getByTestId("Logo") as HTMLImageElement;

  // Asserts Logo component is of type IMG
  expect(renderedLogo.tagName).toBe("IMG");

  // Asserts alt text is "logo"
  expect(renderedLogo.alt).toBe("logo");
});
