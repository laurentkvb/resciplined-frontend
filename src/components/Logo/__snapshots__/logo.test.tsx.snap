// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshot test 1`] = `
<img
  alt="logo"
  data-testid="Logo"
  src="logo.svg"
/>
`;
